sink/signalfx.go

```diff
@@ -173,7 +173,13 @@ func (s *SignalFx) Send(ctx context.Context, m *blip.Metrics) error {
 
 	// Send metrics to SFX. The SFX client handles everything; we just pass
 	// it data points.
-	return s.sfxSink.AddDatapoints(ctx, dp[0:n])
+	err := s.sfxSink.AddDatapoints(ctx, dp[0:n])
+	if err != nil {
+		blip.Debug("error sending data points to SignalFx: %s", err)
+		s.sfxSink.Client.CloseIdleConnections()
```

Comment:
I've not seen this used before. I'm not sure that an error means we need to close idle conns, but I also don't see how it'd hurt. Maybe too many timeouts create too many conns, which become idle?

Reply:
I tried testing without close idle connections and it turned out sfx had no data points for 10+ min due to errors. Our old monitoring has such fix as well for this similar issue.